entities/User.js

```diff
@@ -11,70 +11,75 @@ const findUserByUsername = async (username) => {
 };
 
 const findUserByEmail = async (email) => {
-    const user = await User.findOne({ email });
-    return user;
+  const user = await User.findOne({ email });
+  return user;
 };
 
 const findUserByToken = async (token) => {
-    const user = await User.findOne({ token })
-    .populate('registrations.group')
-    .populate({ //get populate from sub sub document
-      path: "registrations.group", // 1st level subdoc
-      populate: { // 2nd level subdoc 
-        path: "sport",
-        select: "label"
-      }
-    });
-    return user;
-}
+  const user = await User.findOne({ token })
+  .populate('registrations.group')
+  .populate({ //get populate from sub sub document
+    path: "registrations.group", // 1st level subdoc
+    populate: { // 2nd level subdoc 
+      path: "sport",
+      select: "label"
+    }
+  });
+  return user;
+};
+
+const findUsersByGroupId = async(group_id) => {
+  const usersData = await User.find({ 'registrations.group': group_id });
+  return usersData;
+};
 
 const updateUserByToken = async (token, gender, photo, birthDate, description, favoriteSports) => {
-    const updatedUser = await User.updateOne(
-        { token },
-        {
-            gender,
-            photo,
-            birthDate,
-            description,
-            favoriteSports,
-        }
-    );
-    return updatedUser;
+  const updatedUser = await User.updateOne(
+      { token },
+      {
+          gender,
+          photo,
+          birthDate,
+          description,
+          favoriteSports,
+      }
+  );
+  return updatedUser;
 };
 
 const findUserByTokenAndGroupId = async (token, group_id) => {
-    const user = await User.findOne({ token, 'registrations.group': group_id });
-    return user;
+  const user = await User.findOne({ token, 'registrations.group': group_id });
+  return user;
 };
 
-const addUserGroupSubscription = async (token, group_id, status) => {
-    const user = await User.updateOne(
-        { token },
-        {
-          $push: {
-            registrations: {
-              group: group_id,
-              status,
-            }
+const addUserToGroup = async (token, group_id, status) => {
+  const user = await User.updateOne(
+      { token },
+      {
+        $push: {
+          registrations: {
+            group: group_id,
+            status,
           }
         }
-    );
-    return user;
+      }
+  );
+  return user;
 };
 
-const removeUserGroupSubscription = async (token, group_id) => {
-    const user = await User.updateOne(
-        { token },
-        {
-          $pull: {
-            registrations: {
-              group: group_id,
-              status: "Approved",
-            }
+const removeUserFromGroup = async (token, group_id) => {
+  const user = await User.updateOne(
+      { token },
+      {
+        $pull: {
+          registrations: {
+            group: group_id,
+            status: "Approved",
```

Comment:
If not approved, you don't want to remove the user ?

Reply:
ok ! when you implement it, you can re evaluate if it makes sense to make a generic function or not